File: rede-bayseana-smile/tutorials/R/hello.R
library(rSMILE)

# License.R is your licensing key
source("License.R")

net <- Network()
net$readFile("VentureBN.xdsl")
net$setEvidence("Forecast", "Moderate")
net$updateBeliefs()
beliefs <- net$getNodeValue("Success")
for (i in 1:length(beliefs)) {
    cat(sprintf("%s = %f\n", net$getOutcomeId("Success", i-1L), beliefs[i]))
}

File: rede-bayseana-smile/tutorials/cpp/hello.cpp
// hello.cpp

#include "/home/josegustavo/Documentos/IC/rede-bayseana/include/smile.h"
#include "/home/josegustavo/Documentos/IC/rede-bayseana/include/smile_license.h"

// g++ hello.cpp -I/home/josegustavo/Documentos/IC/rede-bayseana/include -L/home/josegustavo/Documentos/IC/rede-bayseana/lib -lsmile -o hello

#include <cstdio>

int main()
{
    DSL_errorH().RedirectToFile(stdout);
    DSL_network net;
    int res = net.ReadFile("VentureBN.xdsl");
    if (DSL_OKAY != res)
    {
        return res;
    }

    net.GetNode("Forecast")->Val()->SetEvidence("Moderate");
    net.UpdateBeliefs();
    DSL_node* sn = net.GetNode("Success");
    const DSL_Dmatrix& beliefs = *sn->Val()->GetMatrix();
    const DSL_idArray& outcomes = *sn->Def()->GetOutcomeIds();
    for (int i = 0; i < outcomes.GetSize(); i++)
    {
        printf("%s=%g\n", outcomes[i], beliefs[i]);
    }

    return DSL_OKAY;
}
